core/rawdb/accessors_chain.go

```diff
@@ -1041,17 +1044,20 @@ func PruneBlocks(tx kv.RwTx, blockTo uint64, blocksDeleteLimit int) error {
 
 	for k, _, err := c.Current(); k != nil; k, _, err = c.Next() {
 		if err != nil {
-			return err
+			return existBlocksToPrune, err
 		}
 
 		n := binary.BigEndian.Uint64(k)
```

Comment:
Two main points of this PR:

1. We assume that there is smth to prune by default
2. The only reason to stop pruning is to reach blockTo (not blockDeleteLimit)